Travis builds are failing

Opening an issue to capture what we currently know.

Initial Travis configuration was merged via https://github.com/sky-map-team/stardroid/pull/137.

However, [builds are still failing](https://travis-ci.org/sky-map-team/stardroid/builds/145445267) with the message:

>Tests on test(AVD) - 6.0 failed: Instrumentation run failed due to 'Process crashed.'
com.android.builder.testing.ConnectedDevice > No tests found.[test(AVD) - 6.0] FAILED 
No tests found. This usually means that your test classes are not in the form that your test runner expects (e.g. don't inherit from TestCase or lack @ Test annotations).
:app:connectedDebugAndroidTest FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:connectedDebugAndroidTest'.
> There were failing tests. See the report at: file:///home/travis/build/sky-map-team/stardroid/app/build/reports/androidTests/connected/index.html

Other potential wrinkle is that if any tests rely on Google APIs or Google Play Services, there is currently a problem getting a working Google API emulator running on Travis.  I'm not sure if this impacts stardroid?

If it does, relevant issues are:

* Android-maps-utils - https://github.com/googlemaps/android-maps-utils/issues/284
* Travis - https://github.com/travis-ci/travis-ci/issues/6122

I believe we're stuck waiting on Travis for resolution for this specific problem, see that issue for latest.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,7 @@
 language: android
 jdk: oraclejdk7
-# Don't use the Travis Container-Based Infrastructure - causes memory issues
-sudo: true
+# Use the Travis Container-Based Infrastructure
+sudo: false
 
 cache:
   directories:
@@ -11,16 +11,23 @@ cache:
 env:
   global:
     - ANDROID_API_LEVEL=23
+    - EMULATOR_API_LEVEL=21
     - ANDROID_BUILD_TOOLS_VERSION=23.0.3
     - ANDROID_ABI=armeabi-v7a
-    - ADB_INSTALL_TIMEOUT=10 # minutes (2 minutes by default)
+    - ANDROID_TAG=google_apis
+    - ADB_INSTALL_TIMEOUT=20 # minutes (2 minutes by default)
 
 android:
   components:
+    - tools # to get the new `repository-11.xml`
     - platform-tools
-    - tools
+    - tools # to install Android SDK tools 25.1.x
     - build-tools-$ANDROID_BUILD_TOOLS_VERSION
     - android-$ANDROID_API_LEVEL
+    - android-$EMULATOR_API_LEVEL
+    # For Google APIs
+    - addon-google_apis-google-$ANDROID_API_LEVEL
+    - addon-google_apis-google-$EMULATOR_API_LEVEL
     # Google Play Services
     - extra-google-google_play_services
     # Support library
@@ -29,11 +36,12 @@ android:
     - extra-google-m2repository
     - extra-android-m2repository
     # Specify at least one system image
-    - sys-img-armeabi-v7a-android-$ANDROID_API_LEVEL
+    - sys-img-armeabi-v7a-google_apis-$ANDROID_API_LEVEL
+    - sys-img-armeabi-v7a-google_apis-$EMULATOR_API_LEVEL
 
 before_script:
   # Create and start emulator
-  - echo no | android create avd --force -n test -t android-$ANDROID_API_LEVEL --abi $ANDROID_ABI
+  - echo no | android create avd --force -n test -t "android-"$EMULATOR_API_LEVEL --abi $ANDROID_ABI --tag $ANDROID_TAG
   - emulator -avd test -no-skin -no-audio -no-window &
   - android-wait-for-emulator
 

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -46,6 +46,12 @@ android {
         // TODO(jontayler): fix the missing translations.
         abortOnError false
     }
+
+    // This enables long timeouts required on slow environments, e.g. Travis
+    adbOptions {
+        timeOutInMs 10 * 60 * 1000  // 10 minutes
+        installOptions "-d", "-t"
+    }
 }
 
 dependencies {

